### PR TITLE
Feature/tournament start end dates

### PR DESCRIPTION
**Pull request by hburn7**

### CODE REVIEW

File: API/DTOs/TournamentDTO.cs [hunk 1]
```diff
@@ -51,6 +51,16 @@ public class TournamentDTO
     [SuppressMessage("ReSharper", "UnusedMember.Global")]
     public DateTime Created { get; set; }
 
+    /// <summary>
+    /// The date of the first match of the tournament
```
Comment (myssto) [inline API/DTOs/TournamentDTO.cs:55]:
```suggestion
    /// The start date of the first match played in the tournament
```

File: API/DTOs/TournamentDTO.cs [hunk 2]
```diff
@@ -51,6 +51,16 @@ public class TournamentDTO
...
+    /// </summary>
+    public DateTime StartTime { get; set; }
+
+    /// <summary>
+    /// The date of the last match of the tournament
```
Comment (myssto) [inline API/DTOs/TournamentDTO.cs:60]:
```suggestion
    /// The end date of the last match played in the tournament
```

File: DataWorkerService/Services/Implementations/OsuApiDataParserService.cs [hunk 3]
```diff
@@ -38,8 +38,6 @@ public async Task ParseMatchAsync(Match match, MultiplayerMatch apiMatch)
             match.Games.Add(newGame);
         }
 
-        match.EndTime = DetermineMatchEndTime(apiMatch) ?? default;
```
Comment (myssto) [inline DataWorkerService/Services/Implementations/OsuApiDataParserService.cs:41]: Why was this removed? No match end times will be populated if this is removed.

File: Database/Entities/Tournament.cs [hunk 4]
```diff
@@ -97,6 +97,16 @@ public class Tournament : UpdateableEntityBase, IProcessableEntity, IAuditableEn
     /// </summary>
     public User? VerifiedByUser { get; set; }
 
+    /// <summary>
+    /// The date of the first match of the tournament
+    /// </summary>
+    public DateTime StartTime { get; set; }
+
+    /// <summary>
+    /// The date of the last match of the tournament
```
Comment (myssto) [inline Database/Entities/Tournament.cs:106]:
```suggestion
    /// The end date of the last <see cref="Match"/> played in the tournament
```

File: Database/Entities/Tournament.cs [hunk 5]
```diff
@@ -97,6 +97,16 @@ public class Tournament : UpdateableEntityBase, IProcessableEntity, IAuditableEn
     /// </summary>
     public User? VerifiedByUser { get; set; }
 
+    /// <summary>
+    /// The date of the first match of the tournament
```
Comment (myssto) [inline Database/Entities/Tournament.cs:101]:
```suggestion
    /// The start date of the last <see cref="Match"/> played in the tournament
```

**Review by myssto — CHANGES_REQUESTED** [review]
This line is necessary for proper parsing, it should not have been removed. Unless there is a problem with it that I didn't know about
```cs
match.EndTime = DetermineMatchEndTime(apiMatch) ?? default;
```

Besides that just some doc string changes

File: DataWorkerService/Services/Implementations/OsuApiDataParserService.cs (hunk 3 above)
Comment (hburn7) [reply DataWorkerService/Services/Implementations/OsuApiDataParserService.cs:41]: I think I deleted this by mistake when I removed some of the code I had here.

Comment (hburn7) [inline DataWorkerService/Services/Implementations/OsuApiDataParserService.cs:41]:
```suggestion
        match.EndTime = DetermineMatchEndTime(apiMatch) ?? default;
```

File: API/DTOs/TournamentDTO.cs (hunk 1 above)
Comment (hburn7) [reply API/DTOs/TournamentDTO.cs:55]: Need to add cref in this like the rest

**Review by myssto — APPROVED** [review]
